### PR TITLE
reimplement template: false to prevent render events

### PR DESCRIPTION
**Pull request by wesmangum**
### Proposed changes
 - Reimplement `template: false` value.
 - Check in view's `render` for falsy template value and will prevent view bindings and any view event triggers.

Link to the issue: https://github.com/marionettejs/backbone.marionette/issues/3484

### CODE REVIEW

File: src/view.js [hunk 1]
```diff
@@ -71,7 +71,9 @@ const View = Backbone.View.extend({
   // Subsequent renders after the first will re-render all nested
   // views.
   render() {
-    if (this._isDestroyed) { return this; }
+    const template = this.getTemplate();
```
Comment (blikblum) [inline src/view.js]: `getTemplate` is being called twice

File: src/mixins/template-render.js [hunk 2]
```diff
@@ -12,12 +11,6 @@ export default {
   _renderTemplate() {
```
Comment (blikblum) [inline src/mixins/template-render.js]: I suggest passing template var as a `_renderTemplate` argument

File: src/mixins/template-render.js [hunk 3]
```diff
@@ -12,12 +11,6 @@ export default {
...
     const template = this.getTemplate();
```
Comment (blikblum) [inline src/mixins/template-render.js]: and remove this call

File: test/unit/view.spec.js [hunk 4]
```diff
@@ -151,8 +131,8 @@ describe('item view', function() {
       expect(this.attachElContentSpy).to.not.have.been.called;
     });
 
-    it('should claim isRendered', function() {
-      expect(this.view.isRendered()).to.be.true;
+    it('should not claim isRendered', function() {
```
Comment (blikblum) [inline test/unit/view.spec.js]: I think we should test also a template: false with a prerendered content. AFAIK isRendered() should be true when the view is created on an el with content

File: src/view.js [hunk 5]
```diff
@@ -71,7 +71,9 @@ const View = Backbone.View.extend({
...
+
+    if (this._isDestroyed || template === false) { return this; }
```
Comment (paulfalgout) [inline src/view.js]: as insignificant as this is, swapping these checks will be slightly more performant for the hotter route.

File: test/unit/view.spec.js [hunk 6]
```diff
@@ -94,48 +94,28 @@ describe('item view', function() {
...
 
-    it('should call bindUIElements', function() {
-      expect(this.bindUIElementsSpy).to.have.been.calledOnce;
+    it('should not call bindUIElements', function() {
+      expect(this.bindUIElementsSpy).to.not.have.been.called;
     });
 
-    it('should trigger a before:render event', function() {
-      expect(this.triggerSpy).to.have.been.calledWith('before:render', this.view);
+    it('should not trigger a before:render event', function() {
```
Comment (paulfalgout) [inline test/unit/view.spec.js]: this test can just be removed.  if the other 2 tests pass this is impossible

File: test/unit/view.spec.js [hunk 7]
```diff
@@ -94,48 +94,28 @@ describe('item view', function() {
...
+      expect(this.triggerSpy).to.not.have.been.calledWith('before:render', this.view);
     });
 
-    it('should trigger a rendered event', function() {
-      expect(this.triggerSpy).to.have.been.calledWith('render', this.view);
+    it('should not trigger a rendered event', function() {
```
Comment (paulfalgout) [inline test/unit/view.spec.js]: this test is also not needed with the previous onRender test I think..

File: test/unit/view.spec.js (hunk 4 above)
Comment (paulfalgout) [reply test/unit/view.spec.js]: Agreed.  Here's an example of that test setup: https://github.com/marionettejs/backbone.marionette/blob/master/test/unit/view.spec.js#L14
You can probably do this without too much extra setup by passing an `el` at instantiation like:
`this.elView = new this.View({ el: '#foo' });`
you just have to make sure `#foo` is in the fixture with content

**Comment by paulfalgout** [comment]
so this would replace https://github.com/marionettejs/backbone.marionette/pull/3470

**Pull request merged**